ddtrace/settings/integration.py

```diff
@@ -32,12 +33,18 @@ def __init__(self, global_config, *args, **kwargs):
         # Set internal properties for this `IntegrationConfig`
         # DEV: By-pass the `__setattr__` overrides from `AttrDict` to set real properties
         object.__setattr__(self, 'global_config', global_config)
+        object.__setattr__(self, 'integration_name', name)
         object.__setattr__(self, 'hooks', Hooks())
         object.__setattr__(self, 'http', HttpConfig())
 
-        # Set default keys/values
+        # Set default analytics configuration, default is disabled
         # DEV: Default to `None` which means do not set this key
-        self['event_sample_rate'] = None
+        # Inject environment variables for integration, override any set in
```

Comment:
Hmm, maybe I had this in the wrong place before, seems like we might want this before the call to `super().__init__()` ?

Since having here will overwrite any set by the init?

e.g.

```python
ic = IntegrationConfig(config, 'molten', analytics=True)
ic is None  # True
```


Right?

Reply:
Or, I commented on this before: https://github.com/DataDog/dd-trace-py/pull/828#discussion_r261326903

Reply:
`.setdefault()` is a nice option here.

Reply:
This comment still applies.